Source/WebCore/platform/network/soup/NetworkStorageSessionSoup.cpp

```diff
@@ -70,14 +71,52 @@ NetworkStorageSession::~NetworkStorageSession()
     g_signal_handlers_disconnect_matched(m_cookieStorage.get(), G_SIGNAL_MATCH_DATA, 0, 0, nullptr, nullptr, this);
 }
 
-void NetworkStorageSession::cookiesDidChange(NetworkStorageSession* session)
+void NetworkStorageSession::cookiesDidChange(NetworkStorageSession* session, SoupCookie* oldCookie, SoupCookie* newCookie, SoupCookieJar*)
 {
     if (session->m_cookieObserverHandler)
```

Comment:
I was a little surprised to see that you don't handle oldCookie and newCookie being set at the same time. Might as well document this assumption:

```suggestion
    ASSERT(!oldCookie || !newCookie);

    if (session->m_cookieObserverHandler)
```

Reply:
This is on purpose, that case is a changed cookie, which the observer still needs to know about.

Reply:
Your current code will call `observer.cookiesAdded` and then return without calling `observer.cookiesDeleted`. You're saying this is intentional? It looks like a bug, so might want to add a clarification.

Reply:
Yes, added a comment.